File: knowledge/public/templates/mobile-sample-apps/android-webview-handoff-skeleton/app/src/main/java/com/example/mobile/session/SessionRepository.kt
package com.example.mobile.session

class SessionRepository {
    data class CookieRecord(
        val name: String,
        val value: String,
        val domain: String,
        val secure: Boolean = true,
        val httpOnly: Boolean = true
    )

    var currentUrl: String = "https://example.mobile.app/webview/login"
        private set
    var currentOrigin: String? = "https://example.mobile.app"
        private set
    var cookies: List<CookieRecord> = emptyList()
        private set
    var localStorage: Map<String, String> = emptyMap()
        private set
    var sessionStorage: Map<String, String> = emptyMap()
        private set
    var headers: Map<String, String> = emptyMap()
        private set

    fun rememberAuthenticatedState(
        targetUrl: String,
        origin: String?,
        cookies: List<CookieRecord>,
        localStorage: Map<String, String>,
        sessionStorage: Map<String, String>,
        headers: Map<String, String>
    ) {
        this.currentUrl = targetUrl
        this.currentOrigin = origin
        this.cookies = cookies
        this.localStorage = localStorage
        this.sessionStorage = sessionStorage
        this.headers = headers
    }
}

File: knowledge/public/templates/mobile-sample-apps/ios-webview-handoff-skeleton/SessionRepository.swift
import Foundation

final class SessionRepository: ObservableObject {
    @Published private(set) var currentURL = URL(string: "https://example.mobile.app/webview/login")!
    @Published private(set) var currentOrigin: String? = "https://example.mobile.app"
    @Published private(set) var cookies: [KyberionCookie] = []
    @Published private(set) var localStorage: [String: String] = [:]
    @Published private(set) var sessionStorage: [String: String] = [:]
    @Published private(set) var headers: [String: String] = [:]

    func rememberAuthenticatedState(
        url: URL,
        origin: String?,
        cookies: [KyberionCookie],
        localStorage: [String: String],
        sessionStorage: [String: String],
        headers: [String: String]
    ) {
        currentURL = url
        currentOrigin = origin
        self.cookies = cookies
        self.localStorage = localStorage
        self.sessionStorage = sessionStorage
        self.headers = headers
    }
}
